src/software/ai/navigator/path_planner/trajectory_path.cpp

```diff
@@ -0,0 +1,111 @@
+#include "software/ai/navigator/path_planner/trajectory_path.h"
+
+#include "software/logger/logger.h"
+
+TrajectoryPath::TrajectoryPath(const BangBangTrajectory2D& initial_trajectory)
+    : traj_path({TrajectoryPathNode(initial_trajectory)})
+{
+}
+
+void TrajectoryPath::append(const KinematicConstraints& constraints,
+                            double connection_time_sec, const Point& destination)
+{
+    for (size_t i = 0; i < traj_path.size(); i++)
+    {
+        if (connection_time_sec <= traj_path[i].getTrajectoryEndTime())
+        {
+            traj_path[i].setTrajectoryEndTime(connection_time_sec);
+
+            // Delete all trajectory nodes after the on that is being connected to
```

Comment:
"... after the node that the current trajectory is connecting to"

Reply:
Updated